java/com/google/flatbuffers/Table.java

```diff
@@ -56,9 +56,11 @@ protected CharsetDecoder initialValue() {
    * @param vtable_offset An `int` offset to the vtable in the Table's ByteBuffer.
    * @return Returns an offset into the object, or `0` if the field is not present.
    */
-  protected int __offset(int vtable_offset) {
-    int vtable = bb_pos - bb.getInt(bb_pos);
-    return vtable_offset < bb.getShort(vtable) ? bb.getShort(vtable + vtable_offset) : 0;
+  protected int __offset(int vtable_offset) { return __offset(vtable_offset, bb_pos - bb.getInt(bb_pos), bb, false); }
```

Comment:
Since this is so performance sensitive, and the new method introduces another conditional, maybe leave the original method as-is.